### PR TITLE
Allow float window when using .rolling

### LINKED ISSUE #12669: ERR: better error message on invalid window when using .rolling
xref #12765. Let's pick up this as a test case and give a more meaningful error message. e.g. for ``Window`` the ``win_type`` is missing, so this should be the error.

```
In [15]: pd.DataFrame(np.arange(10)).rolling(2., center=True).median()
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-15-73c4e557093d> in <module>()
----> 1 pd.DataFrame(np.arange(10)).rolling(2., center=True).median()

/usr/lib/python3.5/site-packages/pandas/core/window.py in median(self, **kwargs)
    811     @Appender(_shared_docs['median'])
    812     def median(self, **kwargs):
--> 813         return super(Rolling, self).median(**kwargs)
    814 
    815     @Substitution(name='rolling')

/usr/lib/python3.5/site-packages/pandas/core/window.py in median(self, how, **kwargs)
    592         if self.freq is not None and how is None:
    593             how = 'median'
--> 594         return self._apply('roll_median_c', how=how, **kwargs)
    595 
    596     _shared_docs['std'] = dedent("""

/usr/lib/python3.5/site-packages/pandas/core/window.py in _apply(self, func, window, center, check_minp, how, **kwargs)
    473             # calculation function
    474             if center:
--> 475                 offset = _offset(window, center)
    476                 additional_nans = np.array([np.NaN] * offset)
    477 

/usr/lib/python3.5/site-packages/pandas/core/window.py in _offset(window, center)
   1356 def _offset(window, center):
   1357     if not com.is_integer(window):
-> 1358         window = len(window)
   1359     offset = (window - 1) / 2. if center else 0
   1360     try:

TypeError: object of type 'float' has no len()
```
Note that `pd.DataFrame(np.arange(10)).rolling(2.).median()` works fine.

pandas 0.18.0.

### PR DESCRIPTION
**Pull request by gliptak**
- [ ] closes #12669
- [ ] tests added
- [ ] passes `git diff upstream/master | flake8 --diff`
- [ ] whatsnew entry

https://github.com/pydata/pandas/issues/12669

### CODE REVIEW

File: pandas/core/window.py [hunk 1]
```diff
@@ -1354,7 +1354,9 @@ def _get_center_of_mass(com, span, halflife, alpha):
 
 
 def _offset(window, center):
-    if not com.is_integer(window):
+    if com.is_float(window):
```
Comment (jreback) [inline pandas/core/window.py]: this needs to raise if its not an integer for Rolling and sub-classes, or int/array of ints for Window

Comment (gliptak) [inline pandas/core/window.py]: @jreback So ```pd.DataFrame(np.arange(10)).rolling(2.).median()``` should work, while ```pd.DataFrame(np.arange(10)).rolling(2., center=True).median()``` should fail with argument validation (only difference is ```center=True```)? Can you confirm?

Comment (jreback) [inline pandas/core/window.py]: neither would wifi now should be an int if not it's an error

**Comment by shoyer** [comment]
I just noted in the associated issue that that I think it would be better to raise than coerce integers.

It could be OK to accept floats if `int(window) == window`, but `window=2.5` should raise an error.

**Comment by jreback** [comment]
I agree we should just raise on non-int. I think was an oversight in the initial impl. yes its safe if ``int(x) == x``, but more correct to raise ``TypeError`` as @shoyer suggests.

**Comment by jreback** [comment]
competing PR is #12718

**Comment by jreback** [comment]
@gliptak want to update (also see my comments on the competeting PR).

**Pull request closed without merging**